target_snowflake/sinks.py

```diff
@@ -23,10 +61,332 @@ def get_sqlalchemy_url(self, config: dict) -> str:
+    # https://docs.snowflake.com/en/sql-reference/sql/alter-table-column.html
+    # TODO: update once https://github.com/meltano/sdk/pull/1114 merges
+    def _adapt_column_type(
+        self,
+        full_table_name: str,
+        column_name: str,
+        sql_type: sqlalchemy.types.TypeEngine,
+    ) -> None:
+        """Adapt table column type to support the new JSON schema type.
+
+        Args:
+            full_table_name: The target table name.
+            column_name: The target column name.
+            sql_type: The new SQLAlchemy type.
+
+        Raises:
+            NotImplementedError: if altering columns is not supported.
+        """
+        current_type: sqlalchemy.types.TypeEngine = self._get_column_type(
+            full_table_name, column_name
+        )
+
+        # Check if the existing column type and the sql type are the same
+        if str(sql_type) == str(current_type):
+            # The current column and sql type are the same
+            # Nothing to do
+            return
+
+        # Not the same type, generic type or compatible types
+        # calling merge_sql_types for assistnace
+        compatible_sql_type = self.merge_sql_types([current_type, sql_type])
+
+        if str(compatible_sql_type) == str(current_type):
+            # Nothing to do
+            return
+
+        if not self.allow_column_alter:
+            raise NotImplementedError(
+                "Altering columns is not supported. "
+                f"Could not convert column '{full_table_name}.{column_name}' "
+                f"from '{current_type}' to '{compatible_sql_type}'."
+            )
+
+        self.connection.execute(
+            sqlalchemy.DDL(
+                "ALTER TABLE %(table)s ALTER COLUMN %(col_name)s SET DATA TYPE %(col_type)s",
+                {
+                    "table": full_table_name,
+                    "col_name": column_name,
+                    "col_type": compatible_sql_type,
+                },
+            )
+        )
+
+    @staticmethod
+    def to_sql_type(jsonschema_type: dict) -> sqlalchemy.types.TypeEngine:
+        """Return a JSON Schema representation of the provided type.
+
+        Uses custom Snowflake types from [snowflake-sqlalchemy](https://github.com/snowflakedb/snowflake-sqlalchemy/blob/main/src/snowflake/sqlalchemy/custom_types.py)
+
+        Args:
+            jsonschema_type: The JSON Schema representation of the source type.
+
+        Returns:
+            The SQLAlchemy type representation of the data type.
+        """
+        # start with default implementation
+        target_type = SQLConnector.to_sql_type(jsonschema_type)
+        # snowflake max and default varchar length
+        # https://docs.snowflake.com/en/sql-reference/intro-summary-data-types.html
+        maxlength = jsonschema_type.get("maxLength", 16777216)
+        # define type maps
+        sting_submaps = [
```

Comment:
Minor typo:

```suggestion
        string_submaps = [
```

And below when referencing the var.